src/models/record.py

```
@@ -16,11 +18,44 @@ def add_phone(self, phone):
     def set_address(self, address):
         self.address = Address(address)
 
+    def add_note(self, title, text):
+        self.notes.append(Note(title, text))
+
+    def find_note(self, title):
+        note_index = self.find_note_index_by_title(title)
+        return self.notes[note_index]
+
+    def remove_note(self, title):
+        self.notes = list(
+            filter(lambda note: note.title.lower() != title.lower(), self.notes)
+        )
+
+    def change_note(self, title, new_text):
+        self.find_note(title).text = new_text
+
+    def change_note_title(self, title, new_title):
+        self.find_note(title).title = new_title
+
+    def add_tag(self, title, tag):
+        self.find_note(title).add_tag(tag)
+
+    def find_note_index_by_title(self, title):
+        index = -1
```

Can we raise an exception here with description?
In this case we'll have the same behavior as Record.find(self, name)
And it'll insure us from the case when find_note(self, title) tries to find inexistent item

---

I think it's better to raise an exception in find_note, because in this function -1 is the correct value, and it's up to you to decide what to do with it.
Regarding the raise exception - so far, we never call find_node without checking for a note, but for future potential use cases, we can add it

---

Indeed. Let's do it then. There can be some potential bugs related to this use case.

---

add to find_note